gradle/plugins/common/src/main/kotlin/junitbuild.spotless-conventions.gradle.kts

```diff
@@ -86,4 +86,9 @@ tasks {
 	named("spotlessMisc") {
 		outputs.doNotCacheIf("negative avoidance savings") { true }
 	}
+	if (System.getenv("spotlessApply") != null && System.getenv("CI") == null) {
+		check {
+			dependsOn("spotlessApply")
```

Comment:
`check` is the wrong lifecycle task to make code changes (same applies for `rewriteRun`) since it needs to run _before_ any compile tasks.

Reply:
agree

Reply:
dependsOn("assemble") should be early enough.

Reply:
- compileJava.dependsOn("rewriteDryRun")
- compileJava.dependsOn("spotlessApply")

Reply:
Thanks. Yes it does, so it will call compile phase itself, before applying the changes.

Reply:
There is no generic compile goal, right, as we now have to impose DRY?

Reply:
If open-rewrite needs the compiled classes we can't make compiling the classes dependent on open-rewrite which means the whole approach of auto-applying its changes and consuming them in the same build doesn't work.

Reply:
yes might not work in the same build, but thats not critical, only some bonus.

We only need to care about `dryRun`. How and if `run` goal then is applied, is non of our business, considering its just done this way currently spot.